src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -85,9 +85,13 @@ public function import($resource, $type = null, $ignoreErrors = false, $sourceRe
     {
         $ret = array();
         $ct = 0;
-        foreach ($this->glob($resource, false, $_, $ignoreErrors) as $resource => $info) {
-            ++$ct;
+        if (!is_string($resource) || false === strpbrk($resource, '*?{[')) {
             $ret[] = $this->doImport($resource, $type, $ignoreErrors, $sourceResource);
+        } else {
+            foreach ($this->glob($resource, false, $_, $ignoreErrors) as $path => $info) {
```

Comment:
Yes, that's how things are now: the Config component doesn't know anything about the container so this cannot trivially be otherwise. I guess $sourceResource is here for that purpose. Not something this PR tries to improve.